runtime/src/accounts_db.rs

```diff
@@ -1311,37 +1303,52 @@ impl AccountsDB {
         accounts_index: &AccountsIndex<AccountInfo>,
         pubkey: &Pubkey,
     ) -> Option<(Account, Slot)> {
-        let (lock, index) = accounts_index.get(pubkey, Some(ancestors), None)?;
-        let slot = lock[index].0;
+        let (slot, store_id, offset) = {
+            let (lock, index) = accounts_index.get(pubkey, Some(ancestors), None)?;
+            let slot_list = lock.slot_list();
+            let slot = slot_list[index].0;
+            let store_id = slot_list[index].1.store_id;
+            let offset = slot_list[index].1.offset;
+            (slot, store_id, offset)
+            // `lock` released here
+        };
+
         //TODO: thread this as a ref
         storage
-            .get_account_storage_entry(slot, lock[index].1.store_id)
+            .get_account_storage_entry(slot, store_id)
             .and_then(|store| {
-                let info = &lock[index].1;
                 store
                     .accounts
-                    .get_account(info.offset)
+                    .get_account(offset)
                     .map(|account| (account.0.clone_account(), slot))
             })
     }
 
     #[cfg(test)]
     fn load_account_hash(&self, ancestors: &Ancestors, pubkey: &Pubkey) -> Hash {
-        let accounts_index = self.accounts_index.read().unwrap();
-        let (lock, index) = accounts_index.get(pubkey, Some(ancestors), None).unwrap();
-        let slot = lock[index].0;
-        let info = &lock[index].1;
+        let (slot, store_id, offset) = {
+            let (lock, index) = self
+                .accounts_index
+                .get(pubkey, Some(ancestors), None)
+                .unwrap();
+            let slot_list = lock.slot_list();
+            let slot = slot_list[index].0;
+            let store_id = slot_list[index].1.store_id;
+            let offset = slot_list[index].1.offset;
```

Comment:
same as this: https://github.com/solana-labs/solana/pull/12787/files?diff=split&w=1#r505171426